src/Helpers.php

```diff
@@ -20,28 +21,87 @@ class Helpers
     public static function makeFolderFilesTree($dir): array
     {
         $tree = [];
-        $ffs = scandir($dir);
-        foreach ($ffs as $ff) {
-            if (!Str::startsWith($ff, '.')) { // We skip hidden folders or config files the directory.
-                $extension = '.' . Str::afterLast($ff, '.');
-                $ff = basename($ff, $extension);
-                $tree[$ff] = [];
-                if (is_dir($dir . '/' . $ff)) {
-                    $tree[$ff] = MaticeServiceProvider::makeFolderFilesTree($dir . '/' . $ff);
+        $localeFolders = File::directories($dir);
+
+        foreach ($localeFolders as $localeFolder) {
+            $locale = basename($localeFolder);
+
+            $tree[$locale] = self::readLocaleFolder($localeFolder);
+        }
+
+        $jsonPaths = self::getJsonPaths($dir);
+
+        foreach ($jsonPaths as $jsonPath) {
+            $files = File::files($jsonPath);
+
+            foreach ($files as $file) {
+                if ($file->getExtension() !== 'json') {
+                    continue;
                 }
-                if (is_file($pathName = $dir . '/' . $ff . $extension)) {
-                    if ($extension === '.json') {
-                        $existingTranslations = $tree[$ff] ?? [];
-                        $tree[$ff] = array_merge(
-                            $existingTranslations,
-                            json_decode(File::get($pathName), true)
-                        );
-                    } else if ($extension === '.php') {
-                        $tree[$ff] = require($pathName);
-                    }
+
+                $locale = $file->getFilenameWithoutExtension();
+
+                $tree[$locale] = array_merge(
+                    $tree[$locale] ?? [],
+                    json_decode($file->getContents(), true)
+                );
+            }
+        }
+
+        return $tree;
+    }
+
+    private static function getJsonPaths($dir): array
+    {
+        $jsonPaths = [];
+
+        if (method_exists(Lang::getLoader(), 'jsonPaths')) {
+            $jsonPaths = Lang::getLoader()->jsonPaths();
+        } elseif (method_exists(Lang::getLoader(), 'getJsonPaths')) {
+            $jsonPaths = Lang::getLoader()->getJsonPaths();
+        }
+
+        return array_unique(
+            array_merge([$dir], $jsonPaths)
+        );
+    }
+
+    private static function readLocaleFolder(string $localeFolder): array
+    {
+        $tree = [];
+        $ffs = scandir($localeFolder);
+
+        foreach ($ffs as $ff) {
+            // We skip hidden folders or config files the directory.
```

Comment:
Small typo introduced by myself 😄 

```suggestion
            // We skip hidden folders or config files in the directory.
```